xarray/tests/test_units.py

```diff
@@ -0,0 +1,1515 @@
+import operator
+from distutils.version import LooseVersion
+
+import numpy as np
+import pytest
+
+import xarray as xr
+from xarray.core.npcompat import IS_NEP18_ACTIVE
+
+pint = pytest.importorskip("pint")
+DimensionalityError = pint.errors.DimensionalityError
+
+
+pytestmark = [
+    pytest.mark.skipif(
+        not IS_NEP18_ACTIVE, reason="NUMPY_EXPERIMENTAL_ARRAY_FUNCTION is not enabled"
+    ),
+    # pytest.mark.filterwarnings("ignore:::pint[.*]"),
+]
+
+
+def use_pint_version_or_xfail(*, version, reason):
+    return pytest.mark.xfail(LooseVersion(pint.__version__) < version, reason=reason)
```

Comment:
Also unused, and in #3358 we went with the philosophy of not polluting the codebase with minimum version tests